crossdock/client/httpserver/behavior.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package httpserver
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	crossdock "github.com/crossdock/crossdock-go"
+	yarpc "github.com/yarpc/yarpc-go"
+	"github.com/yarpc/yarpc-go/crossdock/client/params"
+	"github.com/yarpc/yarpc-go/encoding/raw"
+	"github.com/yarpc/yarpc-go/transport"
+	ht "github.com/yarpc/yarpc-go/transport/http"
+	"golang.org/x/net/context"
+)
+
+// Run exercise a yarpc client against a rigged httpserver.
+func Run(t crossdock.T) {
+	fatals := crossdock.Fatals(t)
+
+	server := t.Param(params.HTTPServer)
+	fatals.NotEmpty(server, "server is required")
+
+	disp := yarpc.NewDispatcher(yarpc.Config{
+		Name: "client",
+		Outbounds: transport.Outbounds{
+			"yarpc-test": ht.NewOutbound(fmt.Sprintf("http://%s:8085", server)),
+		},
+	})
+
+	fatals.NoError(disp.Start(), "could not start Dispatcher")
+	defer disp.Stop()
+
+	runRaw(t, disp)
+}
+
+// runRaw tests if a yarpc client returns a remote timeout error behind the
+// TimeoutError interface when a remote http handler returns a handler timeout.
+func runRaw(t crossdock.T, disp yarpc.Dispatcher) {
+	assert := crossdock.Assert(t)
+	fatals := crossdock.Fatals(t)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+	defer cancel()
+
+	ch := raw.New(disp.Channel("yarpc-test"))
+	_, _, err := ch.Call(yarpc.NewReqMeta(ctx).Procedure("handlertimeout/raw"), nil)
+	fatals.Error(err, "expected an error")
+
+	if transport.IsBadRequestError(err) {
+		t.Skipf("handlertimeout/raw method not implemented: %v", err)
+		return
+	}
+
+	assert.True(transport.IsTimeoutError(err), "returns a TimeoutError: %T", err)
+
+	form := strings.HasPrefix(err.Error(),
```

Comment:
for some reason, existing code was using "form". So I used it, in lack of better idea.